### PR TITLE
add dao feedback

### PR DESCRIPTION
**Pull request by Flip-Liquid**
- reconcile existing add_dao changes with db change
- remove wallet option from onboarding flow, re-add as discrete step
- add ability to tag one thread along after another
- follow add_dao thread with join/onboarding thread
- change onboarding flow to prompt for wallet first
- get/retrieve guild name via cache rather than discord api
- remove dead code

### CODE REVIEW

File: bot/common/threads/guild_select.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import json
```
Comment (Flip-Liquid) [inline bot/common/threads/guild_select.py:1]: moving this file into the threads directory

File: bot/common/threads/utils.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+from bot.common.threads.thread_builder import ThreadKeys, build_cache_value
+from bot.common.threads.onboarding import Onboarding  # noqa: E402
+from bot.common.threads.update import UpdateProfile  # noqa: E402
+from bot.common.threads.initial_contribution import InitialContributions
+from bot.common.threads.report import Report
+from bot.common.threads.points import Points
+from bot.common.threads.add_dao import AddDao
+
+
+async def get_jump_thread(parent_thread, message, user_id):
```
Comment (Flip-Liquid) [inline bot/common/threads/utils.py:12]: moved some common thread manip functions to this common file, rather than having them tied to specific threads.

**Comment by alexkeating** [comment]
Also it looks like the tests are failing

File: bot/common/threads/add_dao.py [hunk 3]
```diff
@@ -53,24 +67,42 @@ async def save(self, message, guild_id, user_id):
             dao_id = str(int(message_content))
         except ValueError:
             message = f"{message_content} is not a valid discord id!"
-            raise ThreadTerminatingException(message)
+            raise ThreadTerminatingException()
```
[the remark this answers is not in the document]
Comment (Flip-Liquid) [reply bot/common/threads/add_dao.py:70]: I removed it when I was trying to debug why these weren't getting caught (due to module import naming) and forgot to re-add it, good catch.

File: bot/common/commands.py [hunk 4]
```diff
@@ -76,33 +72,26 @@ async def report(ctx):
...
-)
-async def join(ctx, wallet):
+async def join(ctx):
     is_guild = bool(ctx.guild)
     if not is_guild:
         raise NotGuildException("Command was executed outside of a guild")
-    if not Web3.isAddress(wallet):
-        await ctx.response.send_message("Not a valid wallet address", ephemeral=True)
-        ctx.response.is_done()
-        return
```
[the remark this answers is not in the document]
Comment (Flip-Liquid) [reply bot/common/commands.py:105]: I just moved it closer to where we actually create the user, line 130 + 131 in onboarding.py

File: bot/common/threads/add_dao.py [hunk 5]
```diff
@@ -53,24 +67,42 @@ async def save(self, message, guild_id, user_id):
...
+                    " You can report your contributions with /report!"
+                )
+                raise ThreadTerminatingException(message)
+
+            # guild exists, user does not, drop into /join flow
+            await write_cache_metadata(user_id, self.cache, "guild_id", dao_id)
+            await write_cache_metadata(user_id, self.cache, "guild_name", guild_name)
+            return StepKeys.ADD_DAO_PREVIOUSLY_ADDED_PROMPT
+
+        await create_guild(dao_id)
```
[the remark this answers is not in the document]
Comment (Flip-Liquid) [reply bot/common/threads/add_dao.py:99]: There was a multi-valued "status" column that was added on the airtable with the values "inputted" "validated" and "onboarded" but I'm not seeing that reflected in the prisma.

**Comment by Flip-Liquid** [comment]
> Also it looks like the tests are failing

fixed, it was due to an incorrect relative import of the exceptions

File: bot/common/threads/add_dao.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Flip-Liquid) [reply bot/common/threads/add_dao.py:99]: likely we should add this to the prisma schema as well. the add dao functinoality is behind a flag, however, so maybe we can take that up in a separate workitem ?

File: bot/common/commands.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Flip-Liquid) [reply bot/common/commands.py:105]: also to allow the /add_dao thread to leverage the same logic

File: bot/common/threads/add_dao.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (alexkeating) [reply bot/common/threads/add_dao.py:99]: Cool, yea as long as it isn't live before that is completed

**Pull request merged**